app/components/autocomplete/emoji_suggestion/emoji_suggestion.tsx

```diff
@@ -0,0 +1,220 @@
+    updateValue: (v: string) => void;
+    onShowingChange: (c: boolean) => void;
+    rootId: string;
+    value: string;
+    nestedScrollEnabled: boolean;
+    skinTone: string;
+    hasFilesAttached: boolean;
+}
+const EmojiSuggestion = ({
+    cursorPosition,
+    customEmojis = [],
+    maxListHeight,
+    updateValue,
+    onShowingChange,
+    rootId,
+    value,
+    nestedScrollEnabled,
+    skinTone,
+    hasFilesAttached,
+}: Props) => {
+    const insets = useSafeAreaInsets();
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const serverUrl = useServerUrl();
+    const flatListStyle = useMemo(() =>
+        [style.listView, {maxHeight: maxListHeight}]
+    , [style, maxListHeight]);
+    const containerStyle = useMemo(() =>
+        ({paddingBottom: insets.bottom + 12})
+    , [insets.bottom]);
+
+    const emojis = useMemo(() => getEmojis(skinTone, customEmojis), [skinTone, customEmojis]);
+
+    const searchTerm = useMemo(() => {
+        const match = value.substring(0, cursorPosition).match(EMOJI_REGEX);
+        return match?.[3] || '';
+    }, [value, cursorPosition]);
+
+    const fuse = useMemo(() => {
+        return new Fuse(emojis, FUSE_OPTIONS);
+    }, [emojis]);
+
+    const data = useMemo(() => {
+        if (searchTerm.length < MIN_SEARCH_LENGTH) {
+            return [];
+        }
+
+        return searchEmojis(fuse, searchTerm);
+    }, [fuse, searchTerm]);
+
+    const showingElements = Boolean(data.length);
+
+    const completeSuggestion = useCallback((emoji: string) => {
+        if (!hasFilesAttached) {
+            const match = value.match(REACTION_REGEX);
+            if (match) {
+                handleReactionToLatestPost(serverUrl, emoji, match[1] === '+', rootId);
+                updateValue('');
+                return;
+            }
+        }
+
+        // We are going to set a double : on iOS to prevent the auto correct from taking over and replacing it
+        // with the wrong value, this is a hack but I could not found another way to solve it
+        let completedDraft: string;
+        let prefix = ':';
+        if (Platform.OS === 'ios') {
+            prefix = '::';
+        }
+
+        const emojiPart = value.substring(0, cursorPosition);
+        const emojiData = getEmojiByName(emoji, customEmojis);
+        if (emojiData?.image && emojiData.category !== 'custom') {
+            const codeArray: string[] = emojiData.image.split('-');
+            const code = codeArray.reduce((acc, c) => {
+                return acc + String.fromCodePoint(parseInt(c, 16));
+            }, '');
+            completedDraft = emojiPart.replace(EMOJI_REGEX_WITHOUT_PREFIX, `${code} `);
+        } else {
+            completedDraft = emojiPart.replace(EMOJI_REGEX_WITHOUT_PREFIX, `${prefix}${emoji}: `);
+        }
+
+        if (value.length > cursorPosition) {
+            completedDraft += value.substring(cursorPosition);
+        }
+
+        updateValue(completedDraft);
```

Comment:
Just for my understanding, when on iOS, this `updateValue` method will be called twice?